File: java/src/main/java/de/oglimmer/mastermind/Console.java
package de.oglimmer.mastermind;

import java.util.Scanner;

public class Console {

    private Scanner scanner = new Scanner(System.in);

    public Guess ask() {
        Guess guess = new Guess();
        while (guess.incomplete()) {
            System.out.print("Enter the order for the 4 colors with spaces (r,g,b,y,o,w): ");
            String s = scanner.nextLine();
            guess.parse(s);
        }
        return guess;
    }
}

File: swift/Sources/MasterMind/Console.swift
//
//  Console.swift
//  MasterMind
//
//  Created by oglimmer on 11/12/21.
//

import Foundation

class Console {
    
    func ask() -> Guess {
        let guess = Guess()
        while (guess.incomplete()) {
            print("Enter the order for the 4 colors with spaces (r,g,b,y,o,w):", terminator: "")
            if let line = readLine() {
                guess.parse(line: line)
            }
        }
        return guess
    }
    
}
